### PR TITLE
Fixed bug causing infinite loops

### PR DESCRIPTION
**Pull request by rvansa**

### CODE REVIEW

**Comment by mcimbora** [comment]
@rvansa Works fine, thanks. However, if 'name' attribute is not specified in the outer loop, NPE is thrown in the last iteration.

See https://gist.github.com/mcimbora/b1e237a9938377f647cd

**Comment by rvansa** [comment]
@mcimbora Does not work fine, then :) Updated.

File: core/src/main/java/org/radargun/stages/control/RepeatEndStage.java [hunk 1]
```diff
@@ -26,17 +26,19 @@ private void updateState(StateBase state) {
...
       state.remove(counterName);
       Stack<String> repeatNames = (Stack<String>) state.get(RepeatStage.REPEAT_NAMES);
-      if (repeatNames == null || repeatNames.isEmpty() || repeatNames.peek() != name) {
+      if (repeatNames == null || repeatNames.isEmpty() || !equals(repeatNames.peek(), name)) {
          throw new IllegalStateException("Cannot unwind repeat: " + repeatNames);
       }
       repeatNames.pop();
    }
 
+   private static boolean equals(String s1, String s2) {
```
Comment (mcimbora) [inline core/src/main/java/org/radargun/stages/control/RepeatEndStage.java]: Wait, this is from Objects.equals, why not using it right away when we support 1.7? :-)

Comment (mcimbora) [inline core/src/main/java/org/radargun/stages/control/RepeatEndStage.java]: Wait, this is from Objects.equals, why not using it right away when we support 1.7? :-)

**Comment by mcimbora** [comment]
@rvansa Integrated, thanks!

**Pull request closed without merging**